scripts/common.py

```diff
@@ -845,7 +845,12 @@ def managed_process(command_args, shell=False, timeout_secs=60, **kwargs):
         for proc in procs_to_kill:
             if proc.is_running():
                 procs_still_alive.append(proc)
-                proc.terminate()
+                try:
+                    proc.terminate()
+                except psutil.NoSuchProcess:
```

Comment:
If we are going to rely on try-except then do we need the `if` check?

Reply:
I think, let's just keep it as-is. If I understand the discussion correctly, there are two possible times when the process might be killed. This helps us distinguish that. If we ever see a "Could not terminate process" message then at least we get some confirmation that this is really a thing. Wdyt?

Reply:
Yeah, I'm fine with that. I have one more question:

is it possible to be in a situation where `proc.is_running() is true` and `proc.terminate() raise NoSuchProcess`?

Reply:
[Sorry I missed this]
Move `procs_still_alive.append(proc)` inside the `try` block`

Reply:
Good catch, but shouldn't it be after the try/except instead?

To your earlier question, I think there is a discussion on the e2e team chat about the possibility of the is_running() check returning True but the process being terminated before terminate() is called.

Reply:
> Good catch, but shouldn't it be after the try/except instead?

I think it should be just after `proc.terminate()` inside `try`, if we keep it after try-except then it will get added even after terminate raise `NoSuchProcess`, right?

>  the possibility of the is_running() check returning True but the process being terminated before terminate() is called.

Hmm.. I'm not sure whether that would be possible for so many failures. But I think it's worth trying! :)